### DISCUSSION TITLE
Configuring MobaXterm or WinSCP to use Teleport? -- Better Options for File Transfer?

### DISCUSSION THREAD

**Discussion by smyoss** [gravitational/teleport#9841] [Q&A]
Hello, Wonderful Teleport Community Folks! 

My team is all in on Teleport and we have our cluster up-and-running as of this week. We're operating on about 20 servers and it's fantastic from a security and usability perspective. We've got VS Code up and running with no issues. However, I have one hiccup which I can't resolve which I was hoping someone might be able to shed some light. My team has used a tool called MobaXterm (https://mobaxterm.mobatek.net/) for years for SSH, VNC, RDP, etc. management. It's fantastic and really integrated into our workflow. Has anybody figured out a way to make this work with Teleport? I know I can configure bash to work with Teleport, but that's not what I'm getting after. I want to be able to use the native SSH functionality to connect to this box. For the life of me, I can't figure it out. I've thought about it a million ways and Googled every possible query I can think of, but with no success. 

Some things I've considered: 
-- I know I can port forward and I've tried that. I can open up port 22 and start a file transfer via SSH, but then I'm back to using keys again. I'm looking to keep Teleport authentication in place 100%. 
-- I've tried to get Moba to recognize the tsh commands, but it won't do it. I've tried everything I can to remap tsh to ssh, but it isn't working. 
-- I've considered adding a local web GUI to manage the file system and access it via port forwarding, but for 20+ instances that would be a lot of work.
-- I've considered opening up FTP and accessing it via port forwarding, but this doesn't seem very secure. 
-- Also, to be clear, I have read every shred of documentation on Teleport's site. 

At the end of the day, the biggest thing I'm looking for is better options for file transfer. The web gui doesn't have enough functionality. VS Code also is lacking on the file transfer abilities. It's super slow and the navigation of the file path isn't very good. I'm really looking for a way to manage large quantities of files coming to and from the infrastructure. The thing that is really nice with MobaXterm is that there is a SCP file browser and a terminal side-by-side. However, any robus file transfer tool which can see the entire instance file system would be option. 

How are y'all managing your files?? Any insight would be GREATLY appreciated. I'm not asking anybody to do the work, but I'm just out of ideas. Anybody that can point me in the right direction would be my hero. 

Gratefully, 

Steve. 

Version -- Teleport v8.1.0 git:v8.1.0-0-g3dc269bef go1.17.3
Clients -- We're all on Windows 10/11. 
MobaXterm -- 21.5
WinSCP -- 5.19.5

![commander](https://user-images.githubusercontent.com/3844721/150087886-4f4c6bb0-4929-4fa3-b4b1-fbcffa7a3ba9.png)
![Cool-features](https://user-images.githubusercontent.com/3844721/150087897-10e35575-7272-4da5-a681-abaaf751776c.png)

**Comment by webvictim** [comment]
**Edit**: PuTTY 0.78+ and WinSCP 6.2+ now fully support Teleport. See docs here: https://goteleport.com/docs/connect-your-client/putty-winscp/

<hr />

Unfortunately neither MobaXTerm nor WinSCP (or basically any graphical SSH client for Windows that I know of) supports SSH certificates, which are the central way that Teleport manages access.

For SSH console via MobaXTerm you could try setting up a PowerShell command rather than using native SSH which just runs the `tsh` utility behind the scenes, similar to this:
![image](https://user-images.githubusercontent.com/897821/152602811-dcc37859-a0f5-4122-8482-9bbf3ff1a5e2.png)

**Reply by webvictim** [reply]
As a (slightly unrelated) update to this, PuTTY version 0.78 (released a few weeks ago) now supports SSH certificates and can be used with Teleport. I'm working on a guide and docs on how to configure this functionality.

Unfortunately it won't address the SCP/SFTP use case, but I'm hoping that WinSCP (which is based on PuTTY) will pick up the SSH certificate support soon and then be able to work with Teleport.

https://winscp.net/forum/viewtopic.php?t=32793
https://winscp.net/tracker/1873

**Reply by bahmanjafari** [reply]
> As a (slightly unrelated) update to this, PuTTY version 0.78 (released a few weeks ago) now supports SSH certificates and can be used with Teleport. I'm working on a guide and docs on how to configure this functionality.
> 
> Unfortunately it won't address the SCP/SFTP use case, but I'm hoping that WinSCP (which is based on PuTTY) will pick up the SSH certificate support soon and then be able to work with Teleport.
> 
> https://winscp.net/forum/viewtopic.php?t=32793 https://winscp.net/tracker/1873

As regards, WinSCP resolved that issue. Is there any update for using WinSCP and Teleport?
In Mobo, I met my needs with the cmd command. But for file transfer, still, some things could be improved via some tools, such as WinSCP or anything like that.

**Reply by webvictim** [reply]
As per https://winscp.net/eng/docs/history It looks like WinSCP 6.0 (which isn't released yet) will use PuTTY 0.78 and therefore include SSH certificate support.

I'm still working on getting support into `tsh` for generating PuTTY configs for automatic connection, but it'll be possible to configure sessions manually before then.

**Reply by abolast** [reply]
> As a (slightly unrelated) update to this, PuTTY version 0.78 (released a few weeks ago) now supports SSH certificates and can be used with Teleport. I'm working on a guide and docs on how to configure this functionality.
> 
> Unfortunately it won't address the SCP/SFTP use case, but I'm hoping that WinSCP (which is based on PuTTY) will pick up the SSH certificate support soon and then be able to work with Teleport.
> 
> https://winscp.net/forum/viewtopic.php?t=32793 https://winscp.net/tracker/1873

Now that winscp v6 is an official version, I'm researching how to configure winscp and teleport to work together.
It is important to realize the GUI use of sftp, the GUI clients I am currently trying are winscp, filezilla and KDE-dolphin.
Please wish me congratulations for being lucky enough to be able to configure it successfully~

**Reply by webvictim** [reply]
Unfortunately https://winscp.net/tracker/2145 (support for SSH host certificates) has still not been implemented, which is essential for WinSCP to be able to work with Teleport.

**Reply by tomynr1** [reply]
Did you find a possibility to configure winscp for usage with teleport?
I made it work with putty, but the settings are not fully imported from my putty sessions to winscp. The main step for putty was configuring the trusted CA, but I don't find this setting in winscp.

**Reply by webvictim** [reply]
Unfortunately it won't work at the moment; it's waiting on WinSCP to add support and implement host certificate verification as per the bug tracker above.

**Comment by Bob565656** [comment]
Thanks! This technique works perfectly
